packages/graphiql/src/components/QueryHistory.js

```diff
@@ -53,7 +58,7 @@ export class QueryHistory extends React.Component {
 
   constructor(props) {
     super(props);
-    this.historyStore = new QueryStore('queries', props.storage);
+    this.historyStore = new QueryStore('queries', props.storage, MAX_HISTORY_LENGTH);
     this.favoriteStore = new QueryStore('favorites', props.storage);
```

Comment:
I just noticed that the favoriteStore doesn't get passed the length, so it can grow unbounded (I think)?

```suggestion
    this.favoriteStore = new QueryStore('favorites', props.storage, MAX_HISTORY_LENGTH);
```

Are there any more QueryStore's around?

Reply:
Favorite is a not LRU cache. From your requirements up in this thread: "Starred queries should never be deleted. If there's only starred queries left we should just give up trying to save new ones." That's my there's not a max length.

There is no other QueryStory.

Reply:
Oh, of course. Let's explicitly pass a value here (null? Infinity?) and maybe add a comment to explain why?

Reply:
Done.